core/src/test/scala/org/apache/spark/executor/ExecutorSuite.scala

```diff
@@ -431,9 +433,10 @@ class ExecutorSuite extends SparkFunSuite
       }
     }
 
-    def errorInGuavaCache(e: => Throwable): Throwable = {
-      val cache = CacheBuilder.newBuilder()
-        .build(new CacheLoader[String, String] {
+    def errorInCaffeinatedGuavaCache(e: => Throwable): Throwable = {
+      val builder = Caffeine.newBuilder()
```

Comment:
From this test, I found that we need to use `CaffeinatedGuava` with  `com.google.common . cache.CacheLoader` to build cache instance to ensure the compatibility of exception behavior of Guava Cache, use `CaffeinatedGuava` with `com.github.benmanes.caffeine.cache.CacheLoader`  or use `Caffeine` directly will make this case fail